refract-spec.md

```diff
@@ -137,9 +133,419 @@ This is how it would expressed in the compact version.
 ["foo", {}, "bar"]
 ```
 
+## Primitive Types
+
+Primitive Types extend upon the Refract Element to create elements for the following types:
+
+- null
+- string
+- number
+- boolean
+- array
+- object
+
+A Refract document MUST support each of these element types by default. These types MAY be extended or replaced based on namespaces provided in any root or parent element.
+
+### Null Element (Element)
+
+A `Null Element` is a type that simply has `null` as its value. It extends upon the `Refract Element` and defines a specific content of `null`.
+
+#### Properties
+- `element`: null (string, fixed)
```

Comment:
So, then is every other primitive type definable using the base element and extend? That should work right and be the definition of them, basically using an id? Like:

{ 
  "element": "extend",
  "attributes": { "id": "string" },
  "content": [
    "element": "element"
  ]
}